### PR TITLE
Updated project dependencies and wrapper

### PR DESCRIPTION
**Pull request by shadygoneinsane**
* TargetSDK and CompileSDK upgraded to API 29
* Migrated to AndroidX
* Added kotlin support
* Added support for data binding

### CODE REVIEW

File: markdeditor/src/main/res/layout/text_component_item.xml [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+        android:layout_height="24dp"
+        android:layout_marginStart="8dp"
         android:layout_marginLeft="8dp"
+        android:layout_marginEnd="0dp"
         android:layout_marginRight="0dp"
-        android:layout_width="24dp"
-        android:layout_height="24dp" />
+        android:gravity="center"
+        android:text="1"
+        android:textColor="#000000"
```
Comment (shadygoneinsane) [inline markdeditor/src/main/res/layout/text_component_item.xml:19]: should move colors to resources

File: markdeditor/src/main/res/layout/text_component_item.xml [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
-        android:textColorHint="#cacaca"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@drawable/blockquote_component_bg"
+        android:paddingStart="4dp"
         android:paddingLeft="4dp"
         android:paddingTop="16dp"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content" />
+        android:textColorHint="#cacaca"
```
Comment (shadygoneinsane) [inline markdeditor/src/main/res/layout/text_component_item.xml:31]: should move colors to resources

File: markdeditor/src/main/java/xute/markdeditor/components/TextComponentItem.java [hunk 3]
```diff
@@ -1,100 +1,101 @@
...
-      indicatorTv.setVisibility(GONE);
-      editText.setBackgroundResource(R.drawable.text_input_bg);
-    } else if (mode == MODE_UL) {
-      indicatorTv.setText(UL_BULLET);
-      indicatorTv.setVisibility(VISIBLE);
-      editText.setBackgroundResource(R.drawable.text_input_bg);
-    } else if (mode == MODE_OL) {
-      indicatorTv.setVisibility(VISIBLE);
-      editText.setBackgroundResource(R.drawable.text_input_bg);
+    public static final int MODE_PLAIN = 0;
```
Comment (shadygoneinsane) [inline markdeditor/src/main/java/xute/markdeditor/components/TextComponentItem.java:19]: Need better Styling constants !
Maybe I'll use @IntDef or Enums

File: app/build.gradle [hunk 4]
```diff
@@ -1,14 +1,15 @@
 apply plugin: 'com.android.application'
```
Comment (shadygoneinsane) [inline app/build.gradle:1]: Should create a separate branch for kotlin + Databinding + AndroidX improvements!

**Comment by bxute** [comment]
Thanks @shadygoneinsane

**Pull request merged**